orttraining/orttraining/training_api/include/onnxruntime_training_c_api.h

```diff
@@ -161,6 +175,36 @@ struct OrtTrainingApi {
   ORT_API2_STATUS(OptimizerStep, _Inout_ OrtTrainingSession* sess,
                   _In_opt_ const OrtRunOptions* run_options);
 
+  /** \brief Registers the use of a linear learning rate scheduler for the training session.
+  *
+  * Register a linear learning rate scheduler that decays the learning rate by linearly updated
+  * multiplicative factor from the initial learning rate set on the training session to 0. The decay
+  * is performed after the initial warm up phase where the learning rate is linearly incremented
+  * from 0 to the initial learning rate set on the training session.
+  *
+  * \param[in] sess The training session that should use the linear learning rate scheduler.
+  * \param[in] warmup_step_count The number of steps in the warm up phase.
+  * \param[in] total_step_count The total number of training steps.
+  *
+  * \snippet{doc} snippets.dox OrtStatus Return Value
+  *
+  */
+  ORT_API2_STATUS(RegisterLinearLRScheduler, _Inout_ OrtTrainingSession* sess, _In_ int64_t warmup_step_count,
```

Comment:
there will be quite a few different scheduler, I see from user repo. do you expect we add RegisterXXX for every one? 

Or mayb we can pass the scheduler type and required parameters with a single register interface?

Reply:
The problem with a single register function is that it will lead to bloated parameters to the function. We might have input parameters that are not relevant to that particular scheduler in the function signature. Even if we put it in a struct, the user might find it confusing to pick the correct arguments for their specific use case.

One solution to this problem is to expose structs specific for each scheduler. For example, to use the LinearLRScheduler, users must pass in a `void*` to a struct of type `LinearLRSchedulerParameters` along with the enum that we expose for `LinearLRScheduler`. We can type cast it to the correct type inside the implementation and accordingly construct the correct C++ object.

Reply:
I made the change so we now take input a `void*` to the learning rate scheduler parameters and an enum. Let me know what you think.